internal/impl/sftp/output.go

```diff
@@ -182,52 +197,68 @@ func (s *sftpWriter) Write(_ context.Context, msg *service.Message) error {
 		flag |= os.O_TRUNC
 	}
 
-	if err := s.client.MkdirAll(filepath.Dir(path)); err != nil {
-		if errors.Is(err, sftp.ErrSshFxConnectionLost) {
-			return service.ErrNotConnected
-		}
-		return err
+	s.sftpClient, err = sftp.NewClient(s.sshClient)
```

Comment:
If we create SFTP client for every write maybe it should be a local variable not a field.

Reply:
Sorted!

Reply:
Thanks.